File: src/TimeMeasure.java
/**
 * Created by Kristofer on 19/02/2017.
 */
public class TimeMeasure {

    private long start = 0;
    private long finish = 0;

    public TimeMeasure() {
    }

    public void start(){
        start = System.nanoTime();
    }

    public void stop(){
        finish = System.nanoTime();
    }

    /**
     * @return measured operation duration in nanoseconds
     */
    public long getDuration(){
        return finish-start;
    }

    public String getDurationString(){
        String result = "";
        long duration = getDuration();
        if (duration > 60000000000L){
            result += Long.toString(duration/60000000000L) + "min ";
            duration = duration % 60000000000L;
        }
        if (duration > 1000000000L){
            result += Long.toString(duration/1000000000L) + "s ";
            duration = duration % 1000000000L;
        }
        result += String.format("%.2f",duration/1000000d).replace(",",".") + "ms";
        return result;
    }
}

File: src/main/sudokusolver/TimeMeasure.kt
package sudokusolver

class TimeMeasure {
    private var start: Long = 0
    private var finish: Long = 0
    fun start() {
        start = System.nanoTime()
    }

    fun stop() {
        finish = System.nanoTime()
    }

    /**
     * @return measured operation duration in nanoseconds
     */
    private val duration: Long
        get() = finish - start

    val durationString: String
        get() {
            var result = ""
            var duration = duration
            if (duration > 60000000000L) {
                result += (duration / 60000000000L).toString() + "min "
                duration %= 60000000000L
            }
            if (duration > 1000000000L) {
                result += (duration / 1000000000L).toString() + "s "
                duration %= 1000000000L
            }
            result += String.format("%.2f", duration / 1000000.0).replace(",", ".") + "ms"
            return result
        }
}